File: Conidae3rd/app/src/main/java/com/example/conidae3rd/MainActivity.java
package com.example.conidae3rd;

import androidx.appcompat.app.AppCompatActivity;

import android.os.Bundle;
import android.widget.Button;

public class MainActivity extends AppCompatActivity {
    private static MainActivity instance = null;
    //↓山岸宅
    //35.71991425470972, 139.9104316936887
    //理科大図書館前(カツシカ)
    //35.77193599712731, 139.8623666081448
    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        //Contextはthisによって渡されるらしい。
        instance = this;
        setContentView(R.layout.activity_main);
        InductionConidae.setGoal(35.77193599712731,139.8623666081448);
        InductionConidae conidae = new InductionConidae();
        Button btnStop = findViewById(R.id.btnStop);
        btnStop.setOnClickListener(view -> conidae.quit());
    }
    //センサの登録やGPSの初期化に必要なオブジェクトであるContextを取得するためのゲッター
    public static MainActivity getInstance() {
        return instance;
    }
}

File: KotlinDae/app/src/main/java/com/example/kotlindae/MainActivity.kt
package com.example.kotlindae

import android.Manifest
import android.content.pm.PackageManager
import android.os.Build
import android.os.Bundle
import android.widget.Toast
import androidx.appcompat.app.AppCompatActivity
import androidx.core.app.ActivityCompat
import androidx.core.content.ContextCompat
import com.example.kotlindae.databinding.ActivityMainBinding

class MainActivity : AppCompatActivity() {
    private lateinit var binding: ActivityMainBinding
    private lateinit var conidae: InductionKonidae
    private lateinit var blue:BluetoothKommunication
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        binding = ActivityMainBinding.inflate(layoutInflater)
        val view = binding.root
        setContentView(view)
        if(allPermissionsGranted()){
            startApplication()
        }
        else{
            ActivityCompat.requestPermissions(
                this, REQUIRED_PERMISSIONS, REQUEST_CODE_PERMISSIONS
            )
        }
        binding.btnStart.setOnClickListener {
            blue = BluetoothKommunication("Shell",this)
            conidae = InductionKonidae(blue,this)
            conidae.setGoals(35.6990494564085, 139.74353592943586)
            }
        binding.btnStop.setOnClickListener {
            conidae.quit()
        }

        binding.btnAction.setOnClickListener {
        }
    }

    override fun onDestroy() {
        super.onDestroy()
        conidae.quit()
    }
    private fun allPermissionsGranted() = REQUIRED_PERMISSIONS.all {
        ContextCompat.checkSelfPermission(
            baseContext, it) == PackageManager.PERMISSION_GRANTED
    }
    private fun startApplication(){

    }
    override fun onRequestPermissionsResult(
        requestCode: Int, permissions: Array<String>, grantResults:
        IntArray) {
        super.onRequestPermissionsResult(requestCode, permissions, grantResults)
        if (requestCode == REQUEST_CODE_PERMISSIONS) {
            if (allPermissionsGranted()) {
                startApplication()
            } else {
                Toast.makeText(this,
                    "Permissions not granted by the user.",
                    Toast.LENGTH_SHORT).show()
                finish()
            }
        }
    }
    companion object{
        private const val TAG = "ConidaeApp"
        private const val REQUEST_CODE_PERMISSIONS = 100
        private val REQUIRED_PERMISSIONS =
            mutableListOf (
                Manifest.permission.CAMERA,
                Manifest.permission.RECORD_AUDIO,
                Manifest.permission.ACCESS_FINE_LOCATION,
                Manifest.permission.ACCESS_COARSE_LOCATION,
                Manifest.permission.BLUETOOTH_CONNECT,
            ).apply {
                if (Build.VERSION.SDK_INT <= Build.VERSION_CODES.P) {
                    add(Manifest.permission.WRITE_EXTERNAL_STORAGE)
                }
            }.toTypedArray()
    }
}